CRM/Contact/BAO/Query.php

```diff
@@ -6751,4 +6754,35 @@ protected function addPseudoconstantFieldToSelect($field, $name, $tableName) {
     $this->_element[$pseudoFieldName] = 1;
   }
 
+  /**
+   * Generate the Group By clause for the query
+   *
+   * @param bool $count
+   * @param bool $sortByChar
+   *
+   * @return string|null
+   */
+  protected function generateGroupByClause($count = FALSE, $sortByChar = FALSE) {
```

Comment:
So if you look at the pr I proposed extracting the getSearchSQLParts function - the idea is that the alphbetQuery would construct it's own 'group by' & we wouldn't need to handle that in the main fn.